Act as a non secure bootloader upon NSEC_BIOS_QEMU

etienne-lms opened the pull request:
Bios_qemu_tz_arm acts as a non secure bootloader **~~when NSEC_BIOS_QEMU is
defined. Bios image location in physical memory depends on earlier bootstage if any~~ when not booted from Qemu special Bios RAM**. When booting in non secure state, bios does not
have to manage the secure world images and boot entry.

Related to:
  https://github.com/OP-TEE/optee_os/pull/2138
  https://github.com/OP-TEE/build/pull/228

(**edited: update description according to latest update**)

jenswikl commented:
bios_qemu_tz_arm is just a hack to get images into memory, while ARM-TF for ARMv7 is an attempt to make a nice example how to boot. Keeping a part of bios_qemu_tz_arm seems to counter that attempt.
If a non-secure boot loader is needed, can't uboot be used instead?

etienne-lms commented:
True: arm-tf / op-tee / u-boot / linux would be a nicer setup.

jenswikl commented:
But is doesn't necessarily be done right now, we can do that in next step.

jenswikl commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

etienne-lms committed:
bios: act a secure or non-secure boot stage depending on boot conditions

Bios_qemu_tz_arm acts as a non secure bootloader when it is executed
from Qemu ROM-like address range (from address 0x0). Secure
initialization are executed only if was booted as primary boot stage
(hence with secure state). Otherwise, initialize only the non secure
side.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Acked-by: Jens Wiklander <jens.wiklander@linaro.org>

etienne-lms commented:
tag applied.
